### PR TITLE
Octopus Tentacle Image

### PR DESCRIPTION
**Pull request by alismx**
## Related Issue

- partial #79

## Changes Proposed

- Creates an initial Tentacle Dockerfile that can be built through our GH actions.
- The image will work as either:
  - Deployment Target: With Octopus Deploy, you can deploy software to Windows servers, Linux servers, Microsoft Azure, AWS, Kubernetes clusters, cloud regions, or an offline package drop. Regardless of where you’re deploying your software, these machines and services are known as your deployment targets.
  - Worker: Workers are machines that can execute tasks that don’t need to be run on the Octopus Server or individual deployment targets.

## Additional Information

- [This page](https://octopus.com/docs/infrastructure/deployment-targets) has useful information about Deployment target licensing
- [This page](https://octopus.com/docs/infrastructure/deployment-targets/tentacle/octopus-tentacle-container) has information on how to run this image as a Deployment Target or a Worker.
- Will need to create an API key
- Seems like the internal port will need to be `10933`
- I'm not sure if setting `TargetWorkerPool` to a value that doesn't exist will create a pool or if it needs to be set to a value predefined in Octopus.

## Testing

- I built the image and did a test run and was able to register the tentacle with our OctopusDeploy instance.
 
### Infrastructure
- [ ] Consult the results of the `terraform-plan` job inside the "Terraform Checks" workflow run for this PR. Confirm that there are no unexpected changes!

### Security
- [ ] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [ ] Any dependencies introduced have been vetted and discussed

## Cloud
- [ ] Oncall has been notified if this change is going in after-hours
- [ ] If there are changes that cannot be tested locally, this has been deployed to our Azure `dev` environment for verification.

### CODE REVIEW

File: octopus/Dockerfile.tentacle [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM octopusdeploy/tentacle:8.1.1693
```
Comment (rin-skylight) [inline octopus/Dockerfile.tentacle:1]: Non-blocking: the newest version is 8.1.1838. These update pretty frequently, but I am also fine leaving it as-is for now. We'll have to come up with an upgrade strategy if/when we ever onboard STLT users to this.

File: .github/actions/build-and-push-octopus/action.yml [hunk 2]
```diff
@@ -35,8 +35,18 @@ runs:
...
-        file: ./octopus/dockerfile
+        file: ./octopus/dockerfile.server
         push: true
         tags: |
           ${{ inputs.acr_registry }}/dibbs-cloud/octopusdeploy:${{ env.VERSION }}
-          ${{ inputs.acr_registry }}/dibbs-cloud/octopusdeploy:latest
\ No newline at end of file
+          ${{ inputs.acr_registry }}/dibbs-cloud/octopusdeploy:latest
+
+    - name: Build and push Docker image
```
Comment (rin-skylight) [inline .github/actions/build-and-push-octopus/action.yml:44]: Can we modify this to indicate that it's the tentacle? Or something else to set it apart from the step outlined at line 34?

File: octopus/Dockerfile.tentacle (hunk 1 above)
Comment (alismx) [reply octopus/Dockerfile.tentacle:1]: I can push this update along with a name update related to your other comment

File: .github/actions/build-and-push-octopus/action.yml (hunk 2 above)
Comment (alismx) [reply .github/actions/build-and-push-octopus/action.yml]: Done.

**Review by shanice-skylight — APPROVED** [review]
Reviewed the files and the terraform plan outputs. The only warning of interest was in the Prod output there are a few`Warning: Value for undeclared variable` warnings. Are these intended to be future variables?

I don't see this as a must to change I was just interested in the intent of the variables. LGTM

**Comment by alismx** [comment]
> Reviewed the files and the terraform plan outputs. The only warning of interest was in the Prod output there are a few`Warning: Value for undeclared variable` warnings. Are these intended to be future variables?
> 
> I don't see this as a must to change I was just interested in the intent of the variables. LGTM

With all the flux in this project, I think they're safe to ignore for now. I'm going to go ahead and merge this.

**Pull request merged**